consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java

```diff
@@ -812,11 +813,7 @@ public KeyedObjectPool<RaftGroup, RatisClient> createClientPool(
         ClientManager<RaftGroup, RatisClient> manager) {
       return new GenericKeyedObjectPool<>(
           new RatisClient.Factory(
-              manager,
-              new ClientFactoryProperty.Builder().build(),
-              properties,
-              clientRpc,
-              MemoizedSupplier.valueOf(() -> config.getImpl())),
+              manager, properties, clientRpc, MemoizedSupplier.valueOf(config::getImpl)),
```

Comment:
`config::getImpl` is computed during initialization, which will construct a closure that includes uninitialized config. we should use `() -> config.getImpl())` here to postpone its computation to the first call.

Reply:
I've handled this by adjusting the initial location of the clientManager ~